### PR TITLE
[14.0] shopfloor : Raise excetion instead dict message

### PR DESCRIPTION
**Pull request by bguillot**
Hi,

This PR replace error messages returned to the app by raises.
It allows  to handle rollback more easily and avoid to take care of the return of methods in case of nested methods. 
For instance if a method returns a recordset but sometimes if error appends it returns dict of error message, so you have to check if the return is dict, and be sure that it will follow all the way down.

I did on one scenario to show the differences. We can change for the others after that.

### CODE REVIEW

**Comment by sebastienbeau** [comment]
@lmignon

**Review by lmignon — APPROVED** [review]
IMO it's a must have... It restore 'normal' behavior when a process encounter an error into a python program.

File: shopfloor_base/__init__.py [hunk 1]
```diff
@@ -2,3 +2,4 @@
 from . import models
 from . import actions
 from . import services
+from . import exceptions
```
Comment (simahawk) [inline shopfloor_base/__init__.py:5]: there's no need to import this file

File: shopfloor_base/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 Akretion (https://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.exceptions import UserError
+
+
+class ShopfloorDispatchError(UserError):
+    def __init__(self, payload):
+        payload["no_wrap"] = True
```
Comment (simahawk) [inline shopfloor_base/exceptions.py:9]: can you document this flag?

File: shopfloor_rest_log/components/service.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    def _dispatch_exception(
+        self, method_name, exception_klass, orig_exception, *args, params=None
+    ):
+        try:
+            super()._dispatch_exception(
+                method_name, exception_klass, orig_exception, *args, params=params
+            )
+        except Exception as e:
+            if isinstance(orig_exception, ShopfloorError):
```
Comment (simahawk) [inline shopfloor_rest_log/components/service.py:22]: can you add a test for this?

**Review by simahawk — CHANGES_REQUESTED** [review]
@bguillot can you rebase?

**Pull request closed without merging**